tests/ModelTester.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <QAbstractItemModel>
+#include <QTest>
+#include <memory>
+
+// does some basic sanity checking (doesn't crash on invalid indexes etc.)
+class ModelTester : public QObject
```

Comment:
It's mainly for development and derps. You usually won't run into any of these issues in the wild.